game-core/src/main/java/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -555,68 +566,24 @@ private static MoveValidationResult validateBasic(final GameData data, final Col
         }
       }
 
-      // Ignore transported units
-      Collection<Unit> moveTest = new ArrayList<>(units);
-      if (route.getStart().isWater()) {
-        moveTest = MoveValidator.getNonLand(units);
-      }
-      final Map<Unit, Collection<Unit>> dependentsMap =
-          getDependents(CollectionUtils.getMatches(units, Matches.unitCanTransport()));
-      final Set<Unit> dependents =
-          dependentsMap.values().stream().flatMap(Collection::stream).collect(Collectors.toSet());
-      dependents.addAll(newDependents.values().stream().flatMap(Collection::stream).collect(Collectors.toSet()));
-      moveTest.removeAll(dependents);
+      // Check that units have enough movement considering land transports
+      final Collection<Unit> unitsWithoutDependents = findNonDependentUnits(units, route, newDependents);
+      final Set<Unit> unitsWithEnoughMovement = unitsWithoutDependents.stream()
+          .filter(unit -> Matches.unitHasEnoughMovementForRoute(route).test(unit)).collect(Collectors.toSet());
```

Comment:
Since you've already moved `.filter(...)` to a new line, it might be easier to read if `.collect(...)` was also on a new line.

Ditto on new line 574 below.

Reply:
Agree. Done.